modules/app-connector/src/server/connection.ts

```diff
@@ -1,137 +1,194 @@
 
-const LOGGER: any = config.logger("app-connector")
+const LOGGER: any = config.logger("app-connector");
 const forge = require("node-forge");
-const keysDirectory = path.resolve("keys")
-const connFile = path.resolve(keysDirectory, "connection.json")
-const crtFile = path.resolve(keysDirectory, "kyma.crt")
-const privateKeyFile = path.resolve(keysDirectory, "app.key")
+const keysDirectory = path.resolve("keys");
+const connFile = path.resolve(keysDirectory, "connection.json");
+const crtFile = path.resolve(keysDirectory, "kyma.crt");
+const privateKeyFile = path.resolve(keysDirectory, "app.key");
+const CronJob = require("cron").CronJob;
 
 var privateKeyData: Buffer;
 var certificateData: Buffer;
 var connection: Info | null = null;
 
-export function init() {
-    if (!fs.existsSync(keysDirectory)) {
-        fs.mkdirSync(keysDirectory)
-    }
-
-    if (fs.existsSync(privateKeyFile)) {
-        privateKeyData = fs.readFileSync(privateKeyFile)
-        LOGGER.info("Found existing private key: %s", privateKeyFile)
-    } else {
-        privateKeyData = generatePrivateKey(privateKeyFile)
-    }
-
-    if (fs.existsSync(connFile)) {
-        connection = JSON.parse(fs.readFileSync(connFile, "utf8"))
-        LOGGER.info("Found existing connection info: %s", connFile)
-    }
+var jobRenewCertificateStarted: Boolean = false;
+var jobRenewCertificatePersistFile: Boolean = false;
+//"0 */1 * * * *" - every minute
+//"00 00 1 * * *" - 1am once a day
+const jobRenewCertificate = new CronJob("00 00 1 * * *", function () {
+  renewCertificate();
+});
 
-    if (fs.existsSync(crtFile)) {
-        certificateData = fs.readFileSync(crtFile)
-        LOGGER.info("Found existing certificate: %s", crtFile)
-    }
+export function init() {
+  if (!fs.existsSync(keysDirectory)) {
+    fs.mkdirSync(keysDirectory);
+  }
+
+  if (fs.existsSync(privateKeyFile)) {
+    privateKeyData = fs.readFileSync(privateKeyFile);
+    LOGGER.info("Found existing private key: %s", privateKeyFile);
+  } else {
+    privateKeyData = generatePrivateKey(privateKeyFile);
+  }
+
+  if (fs.existsSync(connFile)) {
+    connection = JSON.parse(fs.readFileSync(connFile, "utf8"));
+    LOGGER.info("Found existing connection info: %s", connFile);
+  }
+
+  if (fs.existsSync(crtFile)) {
+    certificateData = fs.readFileSync(crtFile);
+    LOGGER.info("Found existing certificate: %s", crtFile);
+  }
+
+  if (privateKeyData && certificateData && connection && connection!.type === Type.Kyma) {
+    LOGGER.info("Found existing files, certificate renewal job will run at %s", jobRenewCertificate.nextDates()._d);
+    jobRenewCertificatePersistFile = true;
+    jobRenewCertificate.start();
+  }
 }
 
 function establish(newConnection: Info, newCertificate: Buffer, persistFiles: boolean) {
-    connection = newConnection
-    certificateData = newCertificate
+  connection = newConnection;
+  certificateData = newCertificate;
+  jobRenewCertificatePersistFile = persistFiles;

```

Comment:
Maybe we should have a dedicated "persisted" flag as part of the connection info structure like we have it for the "insecure" flag.